src/orion/benchmark/task/profet/model_utils.py

```diff
@@ -0,0 +1,483 @@
+        log_C : bool, optional
+            Wether to apply `np.log` onto `C`, by default True.
+        log_Y : bool, optional
+            Wether to apply `np.log` onto `Y`, by default False.
+        n_samples_task : int, optional
+            Number of tasks to sample, by default 500.
+
+        Returns
+        -------
+        Tuple[np.ndarray, np.ndarray, np.ndarray]
+            numpy arrays containing the X, Y, and C's for each task.
+        """
+        n_tasks = Y.shape[0]
+        hidden_space = task_features_std.shape[1]
+        n_configs = X.shape[0]
+
+        X_train = []
+        Y_train = []
+        C_train = []
+
+        for i, xi in enumerate(X):
+            for idx in range(n_tasks):
+                for _ in range(n_samples_task):
+                    multiplier = np.random.randn(hidden_space)
+                    ht = task_features_mean[idx] + task_features_std[idx] * multiplier
+
+                    x = np.concatenate((xi, ht), axis=0)
+                    X_train.append(x)
+                    Y_train.append(Y[idx, i])
+                    C_train.append(C[idx, i])
+
+        X_train = np.array(X_train)
+        Y_train = np.array(Y_train)
+        C_train = np.array(C_train)
+        if log_C:
+            C_train = np.log(C_train)
+
+        if log_Y:
+            Y_train = np.log(Y_train)
+
+        return X_train, Y_train, C_train
+
+    def get_meta_model(
+        self,
+        X_train: np.ndarray,
+        Y_train: np.ndarray,
+        C_train: np.ndarray,
+        with_cost: bool = False,
+    ) -> Tuple[Bohamiann, Optional[Bohamiann]]:
+        """Create, train and return the objective model, and (optionally) a cost model for the data.
+
+        Parameters
+        ----------
+        X_train : np.ndarray
+            Training samples.
+        Y_train : np.ndarray
+            Training objectives.
+        C_train : np.ndarray
+            Training costs.
+        get_architecture : Callable
+            Function used to get a model architecture for a given input dimensionality.
+        config : MetaModelTrainingConfig
+            Configuration options for the training of the meta-model.
+        with_cost : bool, optional
+            Wether to also create a surrogate model for the cost. Defaults to `False`.
+        normalize_targets : bool, optional
+            Wether to normalize the targets (y), by default False.
+
+        Returns
+        -------
+        Tuple[Bohamiann, Optional[Bohamiann]]
+            Surrogate model for the objective, as well as another for the cost, if `with_cose` is True,
+            otherwise `None`.
+        """
+        objective_model = Bohamiann(
+            get_network=type(self).get_architecture,
+            print_every_n_steps=1000,
+            normalize_output=self.normalize_targets,
+        )
+        logger.info("Training Bohamiann objective model.")
+        # NOTE: With the FcNet task, the dataset has size 8_100_000, which takes a LOT of
```

Comment:
I think we can remove this note.

Reply:
Fixed in 4c7073a4c1e89e75358c658411499d615330accb